src/System.Net.Http/tests/FunctionalTests/HttpClientHandlerTest.Http2.cs

```diff
@@ -1597,26 +1599,48 @@ public async Task Http2_PendingSend_SendsReset()
                     request.Content = new StreamContent(stream);
 
                     await Assert.ThrowsAnyAsync<OperationCanceledException>(async () => await client.SendAsync(request, cts.Token));
+
+                    // Send another request to verify that connection is still functional.
+                    request = new HttpRequestMessage(HttpMethod.Get, url);
+                    request.Version = new Version(2,0);
+
+                    await client.SendAsync(request);
                 }
             },
             async server =>
             {
                 Http2LoopbackConnection connection = await server.EstablishConnectionAsync();
 
                 (int streamId, HttpRequestData requestData) = await connection.ReadAndParseRequestHeaderAsync(readBody : false);
-
-                // Cancel client after receiving Headers.
-                cts.Cancel();
+                int frameCount = 0;
                 Frame frame;
                 do
                 {
+                    if (frameCount == (waitForData ? 1 : 0)) {
+                        // Cancel client after receiving Headers or part of request body.
+                        cts.Cancel();
+                    }
                     frame = await connection.ReadFrameAsync(TimeSpan.FromMilliseconds(TestHelper.PassingTestTimeoutMilliseconds)).ConfigureAwait(false);
                     Assert.NotNull(frame); // We should get Rst before closing connection.
                     Assert.Equal(0, (int)(frame.Flags & FrameFlags.EndStream));
+                    frameCount++;
                  } while (frame.Type != FrameType.RstStream);
 
-                 frame = await connection.ReadFrameAsync(TimeSpan.FromMilliseconds(TestHelper.PassingTestTimeoutMilliseconds)).ConfigureAwait(false);
-                 Assert.Null(frame);    // Make sure we do not get any frames after getting RST.
+                 Assert.Equal(1, frame.StreamId);
+
+                frame = null;
+                (streamId, requestData) = await connection.ReadAndParseRequestHeaderAsync();
+                await connection.SendResponseHeadersAsync(streamId, endStream: false, HttpStatusCode.OK);
```

Comment:
These 'await's don't have '.ConfigureAwait(false)'.  But later in this test, some of them do.  Why is this inconsistent?

Reply:
It should not be needed. Test it self is like application. I took some lines from Loopback server e.g. "library" and get extra .ConfigureAwait() from there.